include/nbl/system/CWindowWin32.h

```diff
@@ -1,20 +1,290 @@
 #ifndef __C_WINDOW_WIN32_H_INCLUDED__
 #define __C_WINDOW_WIN32_H_INCLUDED__
 
-#define WIN32_LEAN_AND_MEAN
-#include <windows.h>
-
-#include "nbl/system/IWindow.h"
+#include "nbl/system/IWindowWin32.h"
 
 namespace nbl {
 namespace system
 {
 
-class CWindowWin32 final : public IWindow
+class CWindowWin32 final : public IWindowWin32
 {
-public:
-    using native_handle_t = HWND;
+	// TODO design for event listener etc, WndProc will somehow pass events to provided event listener
+	static LRESULT CALLBACK WndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam)
+	{
```

Comment:
that whole awful thing needs to go in the `.cpp` file

Reply:
right now you could just make it log out "CWindowWin32 event listening unimplemented!"